[EDU-2850] chore: Hexo Boilerplate guide (#430)
aziontech/docs: src/content/docs/pt-br/pages/guias/marketplace/templates/hexo-boilerplate/hexo-boilerplate.mdx

---
title: Como criar edge applications com o Hexo Boilerplate

> description: O Hexo Boilerplate permite que você crie e execute uma Single-Page Application (SPA) baseada em Hexo diretamente no egde

Reviewer: ```suggestion
description: O Hexo Boilerplate permite que você crie e execute uma Single-Page Application (SPA) baseada em Hexo diretamente no egde.
```

> description: O Hexo Boilerplate permite que você crie e execute uma Single-Page Application (SPA) baseada em Hexo diretamente no egde.

Author: ```suggestion
description: O Hexo Boilerplate permite que você crie e execute uma Single-Page Application (SPA) baseada em Hexo diretamente no edge.
```

meta_tags: templates, guides, Azion Marketplace
namespace: docs_guides_hexo_boilerplate
permalink: /documentacao/produtos/guias/hexo-boilerplate/
---

import Badge from '~/components/Badge.astro';

<Badge variant="accent">
Preview
</Badge>

O **Hexo Boilerplate** permite que você crie e execute uma *Single-Page Application (SPA)* baseada em Hexo diretamente no edge, automatizando várias etapas, desde o gerenciamento do repositório até a implantação.

Este template usa a versão `6.3.0` do Hexo.

---

> ## Obter o template

Author: ```suggestion
## Obtenha o template
```

Acesse o [Real-Time Manager (RTM)](https://manager.azion.com/) e selecione a opção **Start with a template** na página inicial.

Isso abrirá uma nova página onde você pode selecionar o card **Hexo Boilerplate**.

---

> ## Configurar o template

Author: ```suggestion
## Configure o template
```

Quando estiver na página do template, clique na aba **Settings** para abrir o formulário de configuração e fornecer as informações para configurar sua aplicação.

Os campos identificados com asterisco são obrigatórios.

- **Application Name** *: o nome da sua aplicação. Ele é usado para identificar e gerenciar sua aplicação na plataforma da Azion.
   - Digite um nome único e fácil de lembrar. Se o nome já tiver sido usado, a plataforma retornará uma mensagem de erro.
- **GitHub Personal Token** *: seu personal token do GitHub.

> :::tip

Author: ```suggestion
:::note
```

Ao gerar seu personal token do GitHub, conceda ao seu **scope** permissões para autorizar um aplicação OAuth ou um personal token a acessar repositórios públicos e privados, incluindo acesso de leitura e reescrita de código. Você também deve habilitar a opção de **workflow** para permitir a adição e atualização de arquivos de workflow de GitHub Actions.

Observe que as permissões podem ter escopo para um usuário ou uma organização ou para um repositório. Leia a documentação sobre a [criação de um personal token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic) para obter mais informações.
:::

Depois de preencher todas as informações, clique no botão **Next**, localizado no canto inferior direito. Isso iniciará o processo de *deployment*.

---

> ## Implantar o template

Author: ```suggestion
## Implante o template
```

Esta página mostra as seguintes seções:

- A **Access your Edge Application**, que inclui o link para visitar e explorar sua aplicação.
- O **Build Summary** contém o nome e a função da aplicação.
- A aba **Deployment details** pode ser aberta para acessar os logs relacionados à implantação.
- **What do you want to do next?** fornece recomendações sobre opções avançadas para aprimorar sua edge application: **Customize Domain**, **Manage Edge Application**, **View Edge Application Metrics**, **View Edge Application Logs** e o botão **Back to Home**.

:::note
O link de sua edge application permite que você veja como ela fica no navegador. No entanto, leva um certo tempo para propagar e configurá-la nas edge locations da Azion. Pode ser necessário aguardar alguns minutos para que a URL seja ativada e para que a página da aplicação seja efetivamente exibida no navegador.
:::

### Principais configurações

Ao implantar este template, por meio da automação, o script cria uma edge application junto com uma edge function e um repositório do GitHub para gerenciar o seu projeto e o código-fonte.

> Seu projeto contem várias configurações, incluindo a configuração da aplicação e seu domínio para hospedá-lo, bem como uma edge function para lógica personalizada.

Reviewer: ```suggestion
Seu projeto contém diversas configurações, incluindo a configuração da aplicação e seu domínio para hospedá-lo, bem como uma edge function para lógica personalizada.
```

O script também confirma os detalhes da implantação no repositório para manter um histórico transparente e rastreável, incluindo informações relacionadas à versão implantada e à data/hora da implantação. Além disso, o repositório também inclui todas as dependências necessárias e uma GitHub Action que, ao ser ativada, garante um fluxo de trabalho de implantação contínuo.

> Para saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#gerenciar-o-template).

Author: ```suggestion
Para saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#gerencie-o-template).
```

> Se você quiser saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#gerenciar-o-template).

Reviewer: ```suggestion
Para saber mais sobre como editar e personalizar as configurações do seu projeto, acesse a seção [Gerenciar o template](#gerenciar-o-template).
```

Além disso, para garantir o desempenho ideal deste template, é necessário ativar:

<Badge>
[Edge Functions](/pt-br/documentacao/produtos/edge-application/edge-functions/)
</Badge>

Você deve ativá-lo separadamente via RTM. Para fazer isso:

1. [Acesse o RTM](https://manager.azion.com/).
2. No canto superior esquerdo, selecione as três linhas horizontais para abrir o **Products menu**.
3. Na seção de **BUILD**, selecione **Edge Application**.
4. Você será redirecionado para a página de **Edge Application**.
- Ela lista todas as edge applications que você criou.
5. Encontre a aplicação relacionada ao seu template e selecione-a.
6. Na aba **Main Settings**, encontre a seção **Edge Application Modules** e ative o switch do produto.

:::caution[Atenção]

> Se este produto estiver ativado, a execução deste template pode gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/pt-br/precos/) para obter mais informações.

Author: ```suggestion
Se este módulo estiver ativado, a execução deste template pode gerar custos relacionados ao uso. Consulte a [página de preços](https://www.azion.com/pt-br/precos/) para obter mais informações.
```

:::

---

> ## Gerenciar o template

Author: ```suggestion
## Gerencie o template
```

Para fazer isso, abra seu novo repositório relacionado com o **Hexo Boilerplate** no GitHub. Em seguida, vá para **Settings** > **Secrets and Variables** > **Action** para [adicionar suas variáveis](https://docs.github.com/en/actions/security-guides/encrypted-secrets), seguindo estas instruções:

1. Adicione o Azion Personal Token aos *secrets*:
- Saiba [como gerar um personal token da Azion](/pt-br/documentacao/produtos/guias/personal-tokens/).

```bash
    AZION_PERSONAL_TOKEN=<value>
```

2. Adicione os ambientes para uso do action workflow no arquivo **main.yml**, incluído na pasta **.github/workflows** do seu repositório:

```yml title=".github/workflows/main.yml"
  - name: edge-...
    id: azion_edge
    ...
    with:
        ....
        azionPersonalToken: ${{ secrets.AZION_PERSONAL_TOKEN }}
        ....

```

3. Abra um pull request para incluir as alterações na main branch e iniciar a implantação automática.

> Agora, seu projeto está pronto para trabalhar com um fluxo de trabalho de implantação contínua, atualizando instantaneamente quaisquer alterações no aplicativo ou no repositório.

Reviewer: ```suggestion
Agora, seu projeto está pronto para trabalhar com um fluxo de trabalho de implantação contínua, atualizando instantaneamente quaisquer alterações na aplicação ou no repositório.
```

> ### Adicionar um domínio personalizado

Author: ```suggestion
### Adicione um domínio personalizado
```
